python/ray/rllib/a3c/a3c.py

```diff
@@ -89,7 +89,7 @@ def compute_gradient(self, params):
 
 class A3CAgent(Agent):
     def __init__(self, env_name, config,
-                 policy_cls=SharedModelLSTM, upload_dir=None):
+                 policy_cls=SharedModel, upload_dir=None):
```

Comment:
Unfortunately the LSTM does not checkpoint, so I reverted this to the basic model for now. @richardliaw

Reply:
What is the specific problem?

Reply:
Actually it seems to be the old problem where we dont support CartPole.

```
[2017-09-26 17:14:28,662] Making new env: CartPole-v0
Traceback (most recent call last):
  File "test/test_checkpoint_restore.py", line 36, in <module>
    alg1 = cls("CartPole-v0", config)
  File "/home/eric/Desktop/ray-private/python/ray/rllib/a3c/a3c.py", line 97, in __init__
    self.env.observation_space.shape, self.env.action_space)
  File "/home/eric/Desktop/ray-private/python/ray/rllib/a3c/shared_model_lstm.py", line 15, in __init__
    super(SharedModelLSTM, self).__init__(ob_space, ac_space, **kwargs)
  File "/home/eric/Desktop/ray-private/python/ray/rllib/a3c/policy.py", line 19, in __init__
    self.setup_graph(ob_space, action_space)
  File "/home/eric/Desktop/ray-private/python/ray/rllib/a3c/shared_model_lstm.py", line 20, in setup_graph
    self._model = LSTM(self.x, self.logit_dim, {})
  File "/home/eric/Desktop/ray-private/python/ray/rllib/models/model.py", line 38, in __init__
    inputs, num_outputs, options)
  File "/home/eric/Desktop/ray-private/python/ray/rllib/models/lstm.py", line 23, in _init
    x = tf.nn.elu(conv2d(x, 32, "l{}".format(i + 1), [3, 3], [2, 2]))
  File "/home/eric/Desktop/ray-private/python/ray/rllib/models/misc.py", line 21, in conv2d
    filter_shape = [filter_size[0], filter_size[1], int(x.get_shape()[3]),
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/tensor_shape.py", line 502, in __getitem__
    return self._dims[key]
IndexError: list index out of range
```

To avoid blocking the release, I think we can just not use the LSTM model for default for now.